### DISCUSSION TITLE
Validate before save returns empty form

### DISCUSSION THREAD

**Discussion by mgdm187** [serenity-is/Serenity#5930] [Help] [answered]
Hi,
I am using a dialog within a dialog and I have some validation rules. In repository, I use `throw new ValidationError("...")` and when this is triggered, the message pops up but it returns an empty form even if it is an update or a create. 
Is there a way to get a message from validation error and closing that dialog or repopulating that form with previous values? Or do I need to create a new button for save and close or something else?

Thank you in advance.
BR

**Comment by VictorTomaili** [comment, marked as answer]
By default when the repository returns a validation error, it doesn't reload the form with a non-entity response. I think you have some custom implementation about filling form values etc. check for your overrides, update UI events, etc. they must call base methods as well or have similar implementations like base methods.
